test/widgets/actions_test.dart

```diff
@@ -269,4 +270,37 @@ void main() {
         expectedMessage: 'NetworkException: Oops (ClientException: Oops)');
     }, skip: true);
   });
+
+  group('updateMessageFlagsStartingFromAnchor', () {
+    String onCompletedMessage(int count) => 'onCompletedMessage($count)';
+    const progressMessage = 'progressMessage';
+    const onFailedTitle = 'onFailedTitle';
+
+    testWidgets('smoke test', (tester) async {
+      final narrow = TopicNarrow.ofMessage(eg.streamMessage());
+      final apiNarrow = narrow.apiEncode()..add(ApiNarrowIsUnread());
+      await prepare(tester);
+      connection.prepare(json: UpdateMessageFlagsForNarrowResult(
+        processedCount: 11, updatedCount: 3,
+        firstProcessedId: null, lastProcessedId: null,
+        foundOldest: true, foundNewest: true).toJson());
+      updateMessageFlagsStartingFromAnchor(
+        context: context, apiNarrow: apiNarrow, op: UpdateMessageFlagsOp.add, flag: MessageFlag.read,
+        includeAnchor: false, startingAnchor: AnchorCode.oldest, onCompletedMessage: onCompletedMessage,
+        onFailedTitle: onFailedTitle, progressMessage: progressMessage);
+      await tester.pump(Duration.zero);
+      check(connection.lastRequest).isA<http.Request>()
+        ..method.equals('POST')
+        ..url.path.equals('/api/v1/messages/flags/narrow')
+        ..bodyFields.deepEquals({
+            'anchor': 'oldest',
+            'include_anchor': 'false',
+            'num_before': '0',
+            'num_after': '1000',
+            'narrow': jsonEncode(apiNarrow),
+            'op': 'add',
+            'flag': 'read',
+          });
```

Comment:
It will be helpful to await the future returned by the previous `updateMessageFlagsStartingFromAnchor` call, and ensure that it returns the expected value, just so we know that it "did pass" as expected.